### PR TITLE
Add isNan and isInf

### PR DESCRIPTION
**Pull request by Alfus**

### CODE REVIEW

File: internal/celext/lib.go [hunk 1]
```diff
@@ -70,6 +71,47 @@ func (l lib) CompileOptions() []cel.EnvOption {
 			l.uniqueMemberOverload(cel.StringType, l.uniqueScalar),
 			l.uniqueMemberOverload(cel.BytesType, l.uniqueBytes),
 		),
+		cel.Function("isNan",
+			cel.MemberOverload(
+				"double_is_nan_bool",
+				[]*cel.Type{cel.DoubleType},
+				cel.BoolType,
+				cel.FunctionBinding(func(args ...ref.Val) ref.Val {
```
Comment (elliotmjackson) [inline internal/celext/lib.go:79]: Wouldnt a `UnaryBinding` be better here?

File: internal/celext/lib.go [hunk 2]
```diff
@@ -70,6 +71,47 @@ func (l lib) CompileOptions() []cel.EnvOption {
...
+					return types.Bool(math.IsNaN(num))
+				}),
+			),
+		),
+		cel.Function("isInf",
+			cel.MemberOverload(
+				"double_is_inf_bool",
+				[]*cel.Type{cel.DoubleType},
+				cel.BoolType,
+				cel.FunctionBinding(func(args ...ref.Val) ref.Val {
```
Comment (elliotmjackson) [inline internal/celext/lib.go:93]: Wouldnt a `UnaryBinding` be better here?

File: internal/celext/lib.go [hunk 3]
```diff
@@ -70,6 +71,47 @@ func (l lib) CompileOptions() []cel.EnvOption {
...
+						return types.Bool(false)
+					}
+					return types.Bool(math.IsInf(num, 0))
+				}),
+			),
+			cel.MemberOverload(
+				"double_int_is_inf_bool",
+				[]*cel.Type{cel.DoubleType, cel.IntType},
+				cel.BoolType,
+				cel.FunctionBinding(func(args ...ref.Val) ref.Val {
```
Comment (elliotmjackson) [inline internal/celext/lib.go:105]: Wouldnt a `BinaryBinding` be better here?

Comment (elliotmjackson) [reply internal/celext/lib.go:105]: also, is this function necessary?

File: internal/celext/lib.go [hunk 4]
```diff
@@ -70,6 +71,47 @@ func (l lib) CompileOptions() []cel.EnvOption {
...
+		),
+		cel.Function("isInf",
+			cel.MemberOverload(
+				"double_is_inf_bool",
+				[]*cel.Type{cel.DoubleType},
+				cel.BoolType,
+				cel.FunctionBinding(func(args ...ref.Val) ref.Val {
+					num, ok := args[0].Value().(float64)
+					if !ok {
+						return types.Bool(false)
```
Comment (elliotmjackson) [inline internal/celext/lib.go:96]:
```suggestion
						return types.UnsupportedRefValConversionErr(num)
```

File: internal/celext/lib.go [hunk 5]
```diff
@@ -70,6 +71,47 @@ func (l lib) CompileOptions() []cel.EnvOption {
...
+			),
+			cel.MemberOverload(
+				"double_int_is_inf_bool",
+				[]*cel.Type{cel.DoubleType, cel.IntType},
+				cel.BoolType,
+				cel.FunctionBinding(func(args ...ref.Val) ref.Val {
+					num, nok := args[0].Value().(float64)
+					sign, sok := args[1].Value().(int64)
+					if !nok || !sok {
+						return types.Bool(false)
```
Comment (elliotmjackson) [inline internal/celext/lib.go:109]: types.UnsupportedRefValConversionErr...

File: internal/celext/lib.go (hunk 3 above)
Comment (Alfus) [reply internal/celext/lib.go:105]: If I can't scope it, might as well match the standard api

**Review by elliotmjackson — APPROVED** [review]

**Pull request merged**